packages/core/helper-plugin/lib/src/components/DynamicTable/tests/Table.test.js

```diff
@@ -0,0 +1,186 @@
+  allowRangeSelection = true,
+}) => {
+  return (
+    <Tbody>
+      {rows.map((data, index) => {
+        const isChecked = entriesToDelete.findIndex((id) => id === data.id) !== -1;
+
+        const handleRowSelection = (event) => {
+          onSelectRow({
+            name: data.id,
+            value: !isChecked,
+            index: allowRangeSelection ? index : undefined,
+            isShiftKeyHeld: allowRangeSelection ? event.nativeEvent.shiftKey : undefined,
+          });
+        };
+
+        return (
+          <Tr key={data.id}>
+            {withMainAction && (
+              <Td>
+                <BaseCheckbox
+                  aria-label={`Select ${data.firstname} ${data.lastname}`}
+                  className={isChecked ? 'selected' : ''}
+                  checked={isChecked}
+                  onChange={handleRowSelection}
+                />
+              </Td>
+            )}
+
+            {headers.map(({ key, name }) => {
+              return (
+                <Td key={key}>
+                  <Typography textColor="neutral800">{data[name] || '-'}</Typography>
+                </Td>
+              );
+            })}
+          </Tr>
+        );
+      })}
+    </Tbody>
+  );
+};
+
+const renderTable = (headers, rows, allowRangeSelection = true, withMainAction = true) =>
+  render(
+    <MemoryRouter>
+      <TrackingProvider value={{ uuid: '2479d6d7-2497-478d-8a34-a9e8ce45f8a7' }}>
+        <ThemeProvider theme={lightTheme}>
+          <IntlProvider locale="en" messages={{}} textComponent="span">
+            <Table
+              headers={headers}
+              contentType="users"
+              rows={rows}
+              withMainAction={withMainAction}
+            >
+              <TableRows headers={headers} rows={rows} allowRangeSelection={allowRangeSelection} />
+            </Table>
+          </IntlProvider>
+        </ThemeProvider>
+      </TrackingProvider>
+    </MemoryRouter>
+  );
+
+describe('DynamicTable', () => {
+  const headers = [
+    {
+      name: 'firstname',
+      metadatas: { label: 'Firstname', sortable: false },
+      key: '__firstname_key__',
+    },
+    {
+      name: 'lastname',
+      metadatas: { label: 'Lastname', sortable: false },
+      key: '__lastname_key__',
+    },
+    {
+      name: 'email',
+      metadatas: { label: 'Email', sortable: false },
+      key: '__email_key__',
+    },
+  ];
+
+  const rows = [
+    { id: 1, firstname: 'soup', lastname: 'soup', email: 'soup@strapi.io' },
+    { id: 2, firstname: 'm', lastname: 'frachet', email: 'm@strapi.io' },
+    { id: 3, firstname: 'hicham', lastname: 'ELBSI', email: 'helbsi@strapi.io' },
+    { id: 4, firstname: 'john', lastname: 'doe', email: 'doe@strapi.io' },
+    { id: 5, firstname: 'mary', lastname: 'jane', email: 'jane@strapi.io' },
+  ];
+
+  it('Performs single row item selection/deselection', async () => {
+    const { container } = renderTable(headers, rows);
+
+    expect(container).toMatchSnapshot();
```

Comment:
```suggestion
```
Please don't just snapshot it, instead assert that we have the headers as we expect etc.

Reply:
@joshuaellis Just out of curiosity why is just snapshotting it not good enough.
Since it will check for an exact match and changing the snapshot would mean chancing the snapshot file aka
Aka if the snapshot has all the correct headers we expect etc. Then we are basically testing it 2 times.

Reply:
In theory yes, they're good, but in practice i've (and others) have found they're not. No one reads / checks them and they're not notoriously hard to follow for HTML that's more than but a few lines especially as it brings in _all the classes_ and even then when you use styled-components classNames are like `c12` which will change with _most_ updates from the DS 🙃

If we wanted to implement visual regression testing i'd use something like chromatic because it's much more user friendly 😄

Reply:
Ye fair so it would be good for backend testing since there it does not change except if you tell it to but for frontend it is bad I see thanks for the explanation .

Reply:
> Ye fair so it would be good for backend testing since there it does not change except if you tell it to but for frontend it is bad I see thanks for the explanation .

yeah exactly, i think it's great for testing data-structures e.g. API responses!